macOS install/build fails when using Python 3.13

Jsonnet installation fails with Python 3.13. It works correctly with Python 3.12 and earlier.

**System details**:

- MacOS 15.2 M1 Pro
- Python 3.13.1
- Apple clang version 16.0.0 (clang-1600.0.26.6), Target: arm64-apple-darwin24.2.0

**Error Trace**:
```
% pip install jsonnet
Collecting jsonnet
  Using cached jsonnet-0.20.0.tar.gz (594 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: jsonnet
  Building wheel for jsonnet (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for jsonnet (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [58 lines of output]
      /private/var/folders/t7/qtwps13s27s4jzsfl4f3lsvr0000gn/T/pip-build-env-m3sxurm7/overlay/lib/python3.13/site-packages/setuptools/_distutils/dist.py:270: UserWarning: Unknown distribution option: 'test_suite'
        warnings.warn(msg)
      running bdist_wheel
      running build
      running build_ext
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/desugarer.cpp -o core/desugarer.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/formatter.cpp -o core/formatter.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/libjsonnet.cpp -o core/libjsonnet.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/lexer.cpp -o core/lexer.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/parser.cpp -o core/parser.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/pass.cpp -o core/pass.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/static_analysis.cpp -o core/static_analysis.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/string_utils.cpp -o core/string_utils.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ core/vm.cpp -o core/vm.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/md5/md5.cpp -o third_party/md5/md5.o
      third_party/md5/md5.cpp:343:5: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
        343 |     sprintf(buf+i*2, "%02x", digest[i]);
            |     ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
        274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
            | ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
        218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
            |                                                       ^
      1 warning generated.
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/char_traits.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/char_traits.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/base64.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/base64.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/language.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/language.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/memory_util.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/memory_util.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/format.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/format.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/time.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/time.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/memory_resource.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/memory_resource.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/error.cpp -o third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/error.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/src/c4/yml/parse.cpp -o third_party/rapidyaml/rapidyaml/src/c4/yml/parse.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/src/c4/yml/preprocess.cpp -o third_party/rapidyaml/rapidyaml/src/c4/yml/preprocess.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/src/c4/yml/common.cpp -o third_party/rapidyaml/rapidyaml/src/c4/yml/common.o
      c++ -c -g -O3 -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++17 -fPIC -Iinclude -Ithird_party/md5 -Ithird_party/json -Ithird_party/rapidyaml/rapidyaml/src/ -Ithird_party/rapidyaml/rapidyaml/ext/c4core/src/ third_party/rapidyaml/rapidyaml/src/c4/yml/tree.cpp -o third_party/rapidyaml/rapidyaml/src/c4/yml/tree.o
      third_party/rapidyaml/rapidyaml/src/c4/yml/tree.cpp:947:44: warning: variable 'icount' set but not used [-Wunused-but-set-variable]
        947 |     for(size_t i = src->first_child(node), icount = 0; i != NONE; ++icount, i = src->next_sibling(i))
            |                                            ^
      1 warning generated.
      building '_jsonnet' extension
      creating build/temp.macosx-15.0-arm64-cpython-313/python
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -Iinclude -I/Users/pwt/xxx/include -I/opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13 -c python/_jsonnet.c -o build/temp.macosx-15.0-arm64-cpython-313/python/_jsonnet.o
      python/_jsonnet.c:189:14: error: call to undeclared function 'PyEval_CallObject'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        189 |     result = PyEval_CallObject(ctx->callback, arglist);
            |              ^
      python/_jsonnet.c:189:12: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
        189 |     result = PyEval_CallObject(ctx->callback, arglist);
            |            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      python/_jsonnet.c:229:14: error: call to undeclared function 'PyEval_CallObject'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        229 |     result = PyEval_CallObject(ctx->callback, arglist);
            |              ^
      python/_jsonnet.c:229:12: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
        229 |     result = PyEval_CallObject(ctx->callback, arglist);
            |            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      4 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for jsonnet
Failed to build jsonnet
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (jsonnet)
```

--- a/.github/workflows/build_and_test.yml
+++ b/.github/workflows/build_and_test.yml
@@ -46,12 +46,20 @@ jobs:
     runs-on: "ubuntu-22.04"
     steps:
       - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
       - name: Build
         run: |
-          python setup.py build
+          pip install build
+          python -m build --sdist --wheel
       - name: Test
         run: |
-          python setup.py test
+          python -m venv --upgrade-deps --clear trial &&
+          cd trial &&
+          source bin/activate &&
+          pip install ../dist/jsonnet-*.whl &&
+          python3 ../python/_jsonnet_test.py
 
   cmake_build:
     name: Build and test using CMake

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+requires = ["setuptools >= 72.2"]

